### PR TITLE
clientv3: support connection timeout

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: clientv3/client_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	"time"
+)
+
+func TestDialTimeout(t *testing.T) {
+	donec := make(chan struct{}, 1)
+	go func() {
+		cfg := Config{
+			Endpoints:   []string{"localhost:12345"},
+			DialTimeout: 2 * time.Second}
+		if c, err := New(cfg); c != nil || err == nil {
```
Comment (xiang90) [inline clientv3/client_test.go]: can we try to compare with the exact error? ErrTimeout or something?

**Comment by xiang90** [comment]
seems like i broke the ci in my previous commit?

**Comment by heyitsanthony** [comment]
it's breaking because I added clientv3 to the test script, I'll fix in this PR

File: clientv3/client_test.go (hunk 1 above)
Comment (heyitsanthony) [inline clientv3/client_test.go]: It gives a grpc connection refused error now; without the timeout keeps retrying. Not how to simulate an actual timeout that'll give `grpc.ErrClientConnTimeout` here.

Comment (xiang90) [inline clientv3/client_test.go]: just setup a plain tcp on the other side that blocks on accept? also we should test the connection refuse case too, right?

Comment (heyitsanthony) [inline clientv3/client_test.go]: I tried `net.Listen("tcp", "localhost:12345")` but the connection went through.

Comment (xiang90) [inline clientv3/client_test.go]: hmm... ok. not other solution come to my mind for now. let's keep this as it is for now. maybe leave a todo.

Comment (heyitsanthony) [inline clientv3/client_test.go]: nevermind, it's returning the correct timeout error with refused connections. test fixed

**Comment by xiang90** [comment]
LGTM

**Pull request merged**